applications/asset_tracker/src/cloud_codec/cloud_codec.c

```diff
@@ -461,19 +546,39 @@ int cloud_encode_env_sensors_data(const env_sensor_data_t *sensor_data,
 #if CONFIG_LIGHT_SENSOR
 /* 4 32-bit ints, 3 spaces, NULL */
 #define LIGHT_SENSOR_DATA_STRING_MAX_LEN ((4 * 11) + 3 + 1)
+#define LIGHT_SENSOR_DATA_NO_UPDATE (-1)
 int cloud_encode_light_sensor_data(const struct light_sensor_data *sensor_data,
 				   struct cloud_msg *output)
 {
 	char buf[LIGHT_SENSOR_DATA_STRING_MAX_LEN];
 	u8_t len;
 	struct cloud_channel_data cloud_sensor;
+	struct light_sensor_data send = { .red = LIGHT_SENSOR_DATA_NO_UPDATE,
+					  .green = LIGHT_SENSOR_DATA_NO_UPDATE,
+					  .blue = LIGHT_SENSOR_DATA_NO_UPDATE,
+					  .ir = LIGHT_SENSOR_DATA_NO_UPDATE };
 
 	if ((sensor_data == NULL) || (output == NULL)) {
 		return -EINVAL;
 	}
 
-	len = snprintf(buf, sizeof(buf), "%d %d %d %d", sensor_data->red,
-		       sensor_data->green, sensor_data->blue, sensor_data->ir);
+	if (cloud_is_send_allowed(CLOUD_CHANNEL_LIGHT_RED, sensor_data->red)) {
+		send.red = sensor_data->red;
+	}
+	if (cloud_is_send_allowed(CLOUD_CHANNEL_LIGHT_GREEN,
```

Comment:
Some blank lines missing between the `if`s

Reply:
Little bit surprised checkpatch doesn't complain about missing blank line before/after if, but as long as it passes I guess it's okay wrt NCS code style

Reply:
I agree here, personally I prefer to separate such expressions with an empty line, but as it doesn't seem to be enforced by the conding standard, I just let it be.